java/ql/src/semmle/code/java/dataflow/internal/DataFlowImplConsistency.qll

```diff
@@ -0,0 +1,127 @@
+/**
+ * Provides consistency queries for checking invariants in the language-specific
+ * data-flow classes and predicates.
+ */
+
+private import DataFlowImplSpecific::Private
+private import DataFlowImplSpecific::Public
+private import TaintTrackingUtil
+
+module Consistency {
+  private predicate relevantNode(Node n) {
```

Comment:
Can this be a class instead of a predicate? I think that would be more idiomatic QL.